cloud-arm/src/main/java/com/sequenceiq/cloudbreak/cloud/arm/ArmResourceConnector.java

```diff
@@ -206,66 +229,118 @@
     }
 
     @Override
-    public List<CloudResourceStatus> downscale(AuthenticatedContext ac, CloudStack stack, List<CloudResource> resources, List<CloudInstance> vms) {
+    public Map<String, Map<String, Object>> collectResourcesToRemove(AuthenticatedContext ac, CloudStack stack, List<CloudResource> resources,
+            List<CloudInstance> vms) {
         AzureRMClient client = armClient.getClient(ac.getCloudCredential());
         ArmCredentialView armCredentialView = new ArmCredentialView(ac.getCloudCredential());
         String stackName = armUtils.getStackName(ac.getCloudContext());
-
-        String resourceGroupName = armUtils.getResourceGroupName(ac.getCloudContext());
-        String diskContainer = armStorage.getDiskContainerName(ac.getCloudContext());
-
-        for (CloudInstance instance : vms) {
-            List<String> networkInterfacesNames = new ArrayList<>();
-            List<String> storageProfileDiskNames = new ArrayList<>();
-            String instanceId = instance.getInstanceId();
-            Long privateId = instance.getTemplate().getPrivateId();
-            ArmDiskType armDiskType = ArmDiskType.getByValue(instance.getTemplate().getVolumeType());
-            String attachedDiskStorageName = armStorage.getAttachedDiskStorageName(armStorage.getArmAttachedStorageOption(stack.getParameters()),
-                    armCredentialView, privateId, ac.getCloudContext(), armDiskType);
+        List<Map> runningNetworkInterfaces = Collections.emptyList();
+        double nodeNumber = stack.getGroups().stream().mapToInt(g -> g.getInstances().size()).sum();
+        if (nodeNumber / vms.size() < PUBLIC_ADDRESS_BATCH_RATIO) {
```

Comment:
We can configure when to download network interfaces in one request, or download one by one